go-controller/pkg/ovn/gateway_init.go

```diff
@@ -550,7 +550,7 @@ func (oc *Controller) addNodeLocalNatEntries(node *kapi.Node, mgmtPortMAC string
 	mgmtPortName := types.K8sPrefix + node.Name
 	stdout, stderr, err := util.RunOVNNbctl("--if-exists", "lr-nat-del", types.OVNClusterRouter,
 		"dnat_and_snat", externalIP.String(), "--",
-		"lr-nat-add", types.OVNClusterRouter, "dnat_and_snat",
+		"--may-exist", "lr-nat-add", types.OVNClusterRouter, "dnat_and_snat",
```

Comment:
this was failing with the following error

> W0211 12:04:07.524985      35 ovn.go:983] error creating gateway for node ovn-control-plane: failed to add dnat_and_snat entry for the management port on node ovn-control-plane, stdout: , stderr: "ovn-nbctl: 169.254.3.63, 10.244.0.2: a NAT with this external_ip and logical_ip already exists\n", error: OVN command '/usr/bin/ovn-nbctl --timeout=15 --if-exists lr-nat-del ovn_cluster_router dnat_and_snat 169.254.3.63 -- lr-nat-add ovn_cluster_router dnat_and_snat 169.254.3.63 10.244.0.2 k8s-ovn-control-plane ce:3e:f5:1c:9c:ca' failed: exit status 1

adding the --may-exist solves the problem, hope is ok?

Reply:
The delete works correctly: 
```
[root@ovn-control-plane ~]# ovn-nbctl lr-nat-list ovn_cluster_router
TYPE             EXTERNAL_IP        EXTERNAL_PORT    LOGICAL_IP            EXTERNAL_MAC         LOGICAL_PORT
dnat_and_snat    169.254.1.122                       10.244.0.2            52:60:a5:37:9f:bc    k8s-ovn-worker2
dnat_and_snat    169.254.1.34                        10.244.1.2            ae:f9:70:78:2a:d6    k8s-ovn-control-plane
dnat_and_snat    169.254.2.242                       10.244.2.2            4e:ac:df:e2:48:2e    k8s-ovn-worker
[root@ovn-control-plane ~]# ovn-nbctl lr-nat-del ovn_cluster_router dnat_and_snat 169.254.1.122
[root@ovn-control-plane ~]# ovn-nbctl lr-nat-list ovn_cluster_router
TYPE             EXTERNAL_IP        EXTERNAL_PORT    LOGICAL_IP            EXTERNAL_MAC         LOGICAL_PORT
dnat_and_snat    169.254.1.34                        10.244.1.2            ae:f9:70:78:2a:d6    k8s-ovn-control-plane
dnat_and_snat    169.254.2.242                       10.244.2.2            4e:ac:df:e2:48:2e    k8s-ovn-worker

```
hmmm...

Reply:
ah it happens if you do it in the same transaction:

```
[root@ovn-control-plane ~]# ovn-nbctl lr-nat-del ovn_cluster_router dnat_and_snat 169.254.1.34 -- lr-nat-add ovn_cluster_router dnat_and_snat 169.254.1.34 10.244.1.2 k8s-ovn-control-plane ae:f9:70:78:2a:d6
ovn-nbctl: 169.254.1.34, 10.244.1.2: a NAT with this external_ip and logical_ip already exists
```

may-exist doesnt fix it, because OVN wont commit the change. You need to delete in a separate transaction, then add in another.